erpnext/controllers/transaction_controller.py

```diff
@@ -736,6 +736,11 @@ def set_taxes_and_charges(self):
 		tax_master_doctype = self.meta.get_field("taxes_and_charges").options
 
 		if (self.is_new() or self.is_pos_profile_changed()) and not self.get("taxes"):
+			if self.company and self.get("customer"):
+				# get default tax template from customer master
+				from erpnext.accounts.party import set_taxes
+				self.taxes_and_charges = set_taxes(self.customer, "Customer", self.get("transaction_date") or self.get("posting_date"), self.company)
+				
```

Comment:
`reset_taxes_and_charges` method is used for both Purchasing documents and Selling documents .. this case only handles selling documents. Handle buying documents as well

Reply:
Also pass all the necessary arguments required to determine the correct tax template